src/NClient/NClient.Standalone/ClientProxy/Generation/Interceptors/ClientInterceptor.cs

```diff
@@ -148,32 +153,58 @@ internal class ClientInterceptor<TClient, TRequest, TResponse> : AsyncIntercepto
             }
         }
         
-        private async Task<object?> ExecuteHttpResponseAsync(ITransportNClient<TRequest, TResponse> transportNClient, IRequest request, Type resultType, IResiliencePolicy<TRequest, TResponse>? resiliencePolicy, CancellationToken cancellationToken)
+        private async Task<object?> ExecuteHttpResponseAsync(ITransportNClient<TRequest, TResponse> transportNClient, IRequest request, Type resultType, IResiliencePolicy<TRequest, TResponse>? resiliencePolicy, ICachingAttribute? cachingAttribute = default, CancellationToken cancellationToken = default)
         {
+            if (await _responseCacheWorker.TryGet(request, cancellationToken) is { } cachedResult)
+            {
+                _toolset.Logger?.LogDebug("Response received from cache. Request id: '{requestId}'.", request.Id);
+                return cachedResult;
+            }
+            
             if (resultType == typeof(TResponse))
                 return await transportNClient
                     .GetOriginalResponseAsync(request, resiliencePolicy, cancellationToken)
                     .ConfigureAwait(false);
 
             if (resultType == typeof(IResponse))
-                return await transportNClient
+            {
+                var result = await transportNClient
                     .GetHttpResponseAsync(request, resiliencePolicy, cancellationToken)
                     .ConfigureAwait(false);
+                
+                await _responseCacheWorker.Put(request, result, cachingAttribute, cancellationToken);
```

Comment:
You need to get and save the cache in the transport client. I'll explain why: if you save the cache here, then invalid responses will not be cached, because they will not come here.